File: bindings/kotlin/examples/ChainId.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newDevnet()
        val chainId = client.chainId()
        println("Chain ID: $chainId")
    } catch (e: Exception) {
        e.printStackTrace()
    }
}

File: bindings/swift/examples/chain_id.swift
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import Foundation
import IotaSDK

@main
struct ChainIdExample {
    static func main() async {
        do {
            let client = try GraphQlClient.newDevnet()
            let chainId = try await client.chainId()
            print("Chain ID: \(chainId)")
        } catch {
            print("Error: \(error)")
            exit(1)
        }
    }
}
